### ISSUE TITLE
Feature: add option to make it so that up and down arrow keys don't open menu

### ISSUE THREAD

**Issue by agusterodin** [tailwindlabs/headlessui#96]
The default up and down arrow key to open dropdown behavior messes with a natural keyboard navigation experience for some use cases. An example of a scenario where this doesn't work well and has been implemented differently is the "more" button on the left panel of the twitter web app.

Also, thoughts on having an option to wrap around to first item if going down on last item in menu?

**Comment by RobinMalfait** [comment]
Hey! Thank you for your feature suggestion!
Much appreciated! 🙏

For your first example, I don't really see how it is a problem to implement the arrow keys to open a Menu? What are some of the issues you that you experience? Afaik they don't implement arrow keys in the side navigation anyway so I don't see how it would *break* stuff. Apart from that the Twitter example is not even using an actual menu, some features are not implemented for example home/end or PageUp/PageDown don't jump to the first item.
Last but not least, when pressing `ArrowUp` it is pretty common to open the menu and focus the *last* non disabled element.

For the wrap around behavior, the aria best practices make it optional to implement that. For this particular case we looked at native OS's to see how they did it and there the wrap around functionality is not implement so we did it this way.

Long story short, I am open for feature requests but only if they fix real problems, so I'm curious about your issues for the first question.

**Comment by agusterodin** [comment]
I think the reason I would want this is because I am using your component in scenarios where the element that serves as the dropdown trigger button isn't a traditionally styled "menu list". With a traditionally styled menu list it makes a lot sense to do the arrow button behavior since there is a down arrow graphic in the menu list that indicates the affordance of "dropping down". The behavior that twitter exhibits in its menu was exactly I was trying to achieve, whether it is right or wrong from an accessibility standpoint.

My example is a dropdown in a navbar that uses a profile image as the trigger button. It feels like a slightly different application of the component compared to what you were doing in the example codesandbox. 

If this seems like a waste of time then I apologize. A little bit of behavioral customization later down the line would be awesome but I understand that you want to enforce consistency toward both aria standards and among your suite of components.

<img width="655" alt="Screen Shot 2020-10-14 at 12 07 58 AM" src="https://user-images.githubusercontent.com/10248395/95942836-38031d00-0db2-11eb-9106-1bdcd6552728.png">

**Comment by agusterodin** [comment]
Another place I have seen this is on Google. This dropdown doesn't open with arrow keys.

<img width="808" alt="Screen Shot 2020-10-14 at 12 28 33 AM" src="https://user-images.githubusercontent.com/10248395/95943589-3fc3c100-0db4-11eb-917a-4934163e34c4.png">

Maybe I am making the mistake of thinking of your library as a "triggered dropdown" component vs a "dropdown list". The Twitter and Google examples are of the former. I am also guessing that making a very general purpose "triggered dropdown" would be a lot more complex and outside the scope of what your are trying to achieve with this library.

The other mistake I could be making is assuming that a ui control implementation I see from a "big tech" company should be treated as gospel.

**Comment by RobinMalfait** [comment]
Yeah the Google example is not a real Menu. You have to think about the Menu Button and Menu more like your menu at the top of your screen (on macOS) ` | Chrome | File | Edit | ...` (this is a menu bar with menu buttons and each item is a menu in itself) or the bar on windows at the top with `File | Edit | ...`.

So for consistency reasons I am going to close this. If screen readers see the menu they would expect to use the arrow keys for keyboard navigation. Changing that behaviour would be suboptimal for people using screen readers and other AT's.
